src/crewai/tools/tool_usage.py

```diff
@@ -123,6 +124,17 @@ def _use(
                 tool=calling.tool_name, input=calling.arguments
             )
 
+        tool_method = tool.func or tool.coroutine
+        is_async_tool = asyncio.iscoroutinefunction(tool_method)
+
+        if is_async_tool:
+            if tool.func:
+                # async tool defined using BaseTool class from crewai_tools
+                async_tool_run = tool._run
+            elif tool.coroutine:
+                # async tool defined using @tool decorator from langchain
+                async_tool_run = tool._arun
```

Comment:
Currently when we define an async function using @tool decorator or BaseTools, they get created differently. 

For @tool decorator - if the function is a coroutine, func=None and the function gets added to the coroutine attribute. 

However, for BaseTools, thats not the case, it remains part of the func attribute. To absorb this complexity from the user's perspective, we do it this way so that users can still define their async tools using _run when using BaseTools. There is no change from the user's perspective other than just adding a async while defining the tool.